sqlglot/dialects/snowflake.py

```diff
@@ -48,10 +49,12 @@ def _builder(args: t.List) -> exp.Func:
                     return exp.UnixToTime(this=value, scale=seq_get(args, 1))
                 if not is_float(value.this):
                     return build_formatted_time(exp.StrToTime, "snowflake")(args)
-            if kind == exp.DataType.Type.DATE and not int_value:
-                formatted_exp = build_formatted_time(exp.TsOrDsToDate, "snowflake")(args)
-                formatted_exp.set("safe", safe)
-                return formatted_exp
+
+        if len(args) == 2 and kind == exp.DataType.Type.DATE and not int_value:
```

Comment:
Do we still need this?

```suggestion
        if len(args) == 2 and kind == exp.DataType.Type.DATE:
```

Reply:
If not we can get rid of L36 as well.

Reply:
Not really, it's covered implicitely